features/steps/thens/psets.py

```diff
@@ -0,0 +1,243 @@
+
+        accepted_values['property_names'] = []
+        accepted_values['property_types'] = []
+        accepted_values['data_types'] = []
+
+        for property_def in make_obj(property_set_attr['property_definitions']):
+            accepted_values['property_names'].append(property_def['property_name'])
+            accepted_values['property_types'].append(property_def['property_type'])
+            accepted_values['data_types'].append(property_def['data_type'])
+
+        accepted_values['applicable_entities'] = [s.split('/')[0] for s in make_obj(property_set_attr['applicable_entities'])]
+
+        # in the ifc2x3 data, predefined type restrictions are imposed as:
+        # | | | applicable_type_value                     | | 
+        # | | | {entity}.PredefinedType={predefinedtype}  | |
+        if property_set_attr['applicable_type_value'] and '.PredefinedType=' in property_set_attr['applicable_type_value']:
+            ptype = property_set_attr['applicable_type_value'].split('.PredefinedType=')[1].upper()
+            accepted_values['applicable_entities_with_predefined_types'] = list(zip(
+                accepted_values['applicable_entities'],
+                (ptype for _ in itertools.count())
+            ))
+        else:
+            # in the ifc4 data, predefined type restrictions are imposed as:
+            # | | | applicable_entities                                                     | | 
+            # | | | ['{entity}','{entity}/{predefinedtype}','{entity2}/{predefinedtype2}']  | |
+            accepted_values['applicable_entities_with_predefined_types'] = [((ab[0], ab[1].upper()) if len(ab) == 2 else (ab[0], None)) for ab in (s.split('/') for s in make_obj(property_set_attr['applicable_entities']))]
+
+        return accepted_values
+
+def get_pset_definitions(context, table):
+    schema_specific_path = system.get_abs_path(f"resources/{context.model.schema.upper()}/{table}.csv")
+
+    if os.path.exists(schema_specific_path):
+        tbl_path = schema_specific_path
+    else:
+        tbl_path =system.get_abs_path(f"resources/{table}.csv")
+
+    tbl = system.get_csv(tbl_path, return_type='dict')
+    return {d['property_set_name']: d for d in tbl}
+
+
+
+def get_accepted_pset_values(context, table):
+    return establish_accepted_pset_values(context.activation_attr, get_pset_definitions(context, table))
+
+@gherkin_ifc.step('It must use predefined values according to the "{table}" table')
+def step_impl(context, inst, table):
+    property_set_definitions = get_pset_definitions(context, table)
+
+    if inst not in property_set_definitions.keys():
+        yield ValidationOutcome(inst=inst, observed = {'value':inst}, severity=OutcomeSeverity.ERROR)
+
+
+@gherkin_ifc.step('It must be named according to the property set definitions table "{table}"')
+def step_impl(context, inst, table):
+    accepted_values = get_accepted_pset_values(context, table)
```

Comment:
What is `inst` here? How can we rewrite this without a global state in context.activation_atrr?

Reply:
The name of the each individual property within a property set. 
```
        Given its attribute HasProperties
        Given its attribute Name -> inst 
```

Alternatively we could go back in the stack tree. Probably this would be more reliable and easier to trace down in the future then setting global variables.

Reply:
Or use the activation_inst that we determined when implementing the gcert work. E.g. in 
https://github.com/buildingSMART/ifc-gherkin-rules/blob/4217e02b06a295c73ca2c93125a565fe0151bce1/features/steps/validation_handling.py#L245

We could add it to  
`step_results = list(filter(lambda x: x.severity in [OutcomeSeverity.ERROR, OutcomeSeverity.WARNING], list(fn(context, inst=inst, **kwargs))))`

So 
```
            context.current_activation_inst = activation_inst
            step_results = list(filter(lambda x: x.severity in [OutcomeSeverity.ERROR, OutcomeSeverity.WARNING], list(fn(context, inst=inst, **kwargs))))
            context.current_activation_inst = None
```
Gives access to the relevant pset within the `step_impl` func